BJS2-3514 - Система альбомов постов - Unicorn BC3

hhzktoeto opened the pull request.

hhzktoeto committed:
Создал контроллер, дто, маппер, сервис

hhzktoeto committed:
Merge branch 'feature-BJS2-3491' into feature-BJS2-3514

hhzktoeto committed:
создал метод create и валидатор

hhzktoeto committed:
добавление поста + избранные

hhzktoeto committed:
Мелкие поправки

изменил @RequestHeader на userContext.getUserId()

hhzktoeto committed:
Добавил getUsersAlbums и фильтры

hhzktoeto committed:
Аннотации для Swagger + мелкие поправки

аннотации в AlbumContoller
добавил @RequestBody на AlbumFilterDto в getUsersAlbums

hhzktoeto committed:
Основная логика без тестов

hhzktoeto committed:
Тесты на сервис + мелкие поправки

написал тесты на все методы AlbumService
поправил имя метода deleteAlbumFromFavourites
добавил валидацию validateAlbumTitle в метод update в сервисе

hhzktoeto committed:
Дописал все тесты

тест на валидатор и фильтры

hhzktoeto committed:
Merge branch 'unicorn-master-bc3' into feature-BJS2-3514

hhzktoeto committed:
мелкие поправки

snailrake commented on src/main/java/faang/school/postservice/controller/album/AlbumController.java:
```diff
@@ -0,0 +1,96 @@
+@RequiredArgsConstructor
+@RequestMapping("/albums")
+@Tag(name = "Albums", description = "Endpoints for managing albums")
+public class AlbumController {
```
А где "Пользователь может получить любой альбом по id этого альбома."?

hhzktoeto committed:
changes after cross-review

added getAlbum method
changed deleteAlbumFromFavourites method - now it receives userId and albumId

hhzktoeto committed:
added test

hhzktoeto committed:
minor change
